app.rb

```diff
@@ -33,13 +34,17 @@ def list_people
       prompt
     else
       @people.each_with_index do |person, i|
-        puts "#{i}) [#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+        if person
+          puts "#{i}) [#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+        else
+          puts "#{i}) [Person data is corrupted or incomplete]"
+        end
```

Comment:
- Good job done so far, but when I list the names of people in the `json` file, the names arent listed as expected. Please fix this and ensure the names are listed properly 👍 👍 and ensure your application runs without bugs. I suggest you implement a logic to check class type for the person listed, either Student or Teacher, when reading for the `json` file. 
![ewrgerg](https://github.com/PrinceBadru/OOP-school-library/assets/73607512/fef1c680-bede-4938-bf0e-6815e2c96fd8)